### PR TITLE
Clean up f-strings in logging calls

### PR DESCRIPTION
**Pull request by josh-fell**
This PR cleans up f-string use in logging calls throughout. This is part of initial prep to restrict f-string use in logging calls via CI in the future.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by josh-fell** [comment]
It's not entirely clear to me why there are so many svg file changes from running static checks locally. Is this expected with the new Breeze?

**Comment by potiuk** [comment]
The problem is that you have still old rich in the .pre-commit environment for breeze (and there is a single change in the commands - adding new static checks). Run `breeze self-upgrade`, remove `images/breeze/output-commands-hash.txt` and run `breeze static-check --last-commit`. It should help.

**Comment by josh-fell** [comment]
Thanks for the hot tip @potiuk! I'll try these out.

**Comment by potiuk** [comment]
With new rich there should also be differences in generateda images but far less of those

**Comment by potiuk** [comment]
Still strange that the image generated uses old rich. Did you run `breeze self-upgrade` before running the static check? I guess possibly adding rich>=12.4.1 might be needed.

File: airflow/providers/google/cloud/sensors/dataproc.py [hunk 1]
```diff
@@ -79,8 +79,9 @@ def poke(self, context: "Context") -> bool:
                     job_id=self.dataproc_job_id, region=self.region, project_id=self.project_id
                 )
             except ServerError as err:
-                self.log.info(f"DURATION RUN: {self._duration()}")
-                if self._duration() > self.wait_timeout:
+                duration = self._duration()
+                self.log.info("DURATION RUN: %f")
```
Comment (mik-laj) [inline airflow/providers/google/cloud/sensors/dataproc.py]: Missing parameter.

File: airflow/models/connection.py [hunk 2]
```diff
@@ -208,8 +208,8 @@ def get_uri(self) -> str:
         """Return connection in URI format"""
         if '_' in self.conn_type:
             self.log.warning(
-                f"Connection schemes (type: {str(self.conn_type)}) "
-                f"shall not contain '_' according to RFC3986."
+                "Connection schemes (type: %s) shall not contain '_' according to RFC3986.",
+                str(self.conn_type),
```
Comment (uranusjr) [inline airflow/models/connection.py]:
```suggestion
                self.conn_type,
```

The logging framework stringify things automatically.

File: airflow/models/dagrun.py [hunk 3]
```diff
@@ -790,7 +790,7 @@ def _emit_true_scheduling_delay_stats_for_finished_state(self, finished_tis: Lis
                 if true_delay.total_seconds() > 0:
                     Stats.timing(f'dagrun.{dag.dag_id}.first_task_scheduling_delay', true_delay)
         except Exception as e:
-            self.log.warning(f'Failed to record first_task_scheduling_delay metric:\n{e}')
+            self.log.warning('Failed to record first_task_scheduling_delay metric:\n%s', e)
```
Comment (uranusjr) [inline airflow/models/dagrun.py]: How about

```suggestion
            self.log.warning(
                'Failed to record first_task_scheduling_delay metric.',
                exc_info=True,
            )
```

(Also need to remove `as e` above)

File: airflow/providers/google/cloud/sensors/dataproc.py (hunk 1 above)
Comment (uranusjr) [inline airflow/providers/google/cloud/sensors/dataproc.py]:
```suggestion
                self.log.info("DURATION RUN: %f", duration)
```

File: airflow/providers/google/suite/transfers/sql_to_sheets.py [hunk 4]
```diff
@@ -120,7 +120,7 @@ def execute(self, context: Any) -> None:
             impersonation_chain=self.impersonation_chain,
         )
 
-        self.log.info(f"Uploading data to https://docs.google.com/spreadsheets/d/{self.spreadsheet_id}")
+        self.log.info("Uploading data to https://docs.google.com/spreadsheets/d/%s", self.spreadsheet_id)
```
Comment (uranusjr) [inline airflow/providers/google/suite/transfers/sql_to_sheets.py]: This particular one is tricky; having a URL like this doesn’t really make sense without formatting. Maybe we should do

```suggestion
        if self.log.isEnabledFor("info"):
            url = f"https://docs.google.com/spreadsheets/d/{self.spreadsheet_id}"
            self.log.info("Uploading data to %s", url)
```

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -723,6 +723,14 @@ repos:
...
         exclude: ^airflow/_vendor/
         additional_dependencies: ['rich']
+      - id: check-fstrings-logging
+        name: Ensure f-strings are not used in logging calls.
+        pass_filenames: false
+        always_run: true
+        entry: ./scripts/ci/pre_commit/pre_commit_check_logging_for_fstrings.py
+        language: python
+        require_serial: true
+        additional_dependencies: ['rich']
```
Comment (kaxil) [inline .pre-commit-config.yaml]: You can use this flake8 extension https://github.com/globality-corp/flake8-logging-format . I was planning to do this when I added it for astronomer-providers repo - https://github.com/astronomer/astronomer-providers/blob/main/.pre-commit-config.yaml#L66

```
flake8............................................................................Failed
- hook id: flake8
- exit code: 1

astronomer/providers/amazon/aws/operators/emr.py:64:23: G004 Logging statement uses f-string
```

File: airflow/providers/google/suite/transfers/sql_to_sheets.py (hunk 4 above)
Comment (kaxil) [reply airflow/providers/google/suite/transfers/sql_to_sheets.py]: +1

File: airflow/models/connection.py (hunk 2 above)
Comment (josh-fell) [reply airflow/models/connection.py]: 🤦 Yep, thanks.

File: airflow/models/dagrun.py (hunk 3 above)
Comment (josh-fell) [reply airflow/models/dagrun.py]: Brilliant.

File: airflow/providers/google/cloud/sensors/dataproc.py (hunk 1 above)
Comment (josh-fell) [reply airflow/providers/google/cloud/sensors/dataproc.py]: Good catch @uranusjr and @mik-laj!

File: airflow/providers/google/suite/transfers/sql_to_sheets.py (hunk 4 above)
Comment (josh-fell) [reply airflow/providers/google/suite/transfers/sql_to_sheets.py]: Thanks for the pro tip!

File: .pre-commit-config.yaml (hunk 5 above)
Comment (josh-fell) [reply .pre-commit-config.yaml]: Lovely! I'll remove the custom script here and introduce this extension in later PRs with cleanup.

Comment (potiuk) [reply .pre-commit-config.yaml]: good idea!

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**